[ISSUE] Cannot update the owner of a `databricks_storage_credential`

### Configuration
```hcl
resource "databricks_storage_credential" "foo" {
  aws_iam_role {
    role_arn = var.role_arn
  }

  name  = "foo"
  owner = var.owner
}

resource "databricks_external_location" "audit_logs" {
  credential_name = databricks_storage_credential.foo.id
  name            = "foo"
  owner           = var.owner
  url             = "s3://foo"
}
```

### Expected Behavior
If the value of `owner` changes, the owner of the storage credential should be updated in place.

### Actual Behavior
Terraform errors out with:

```
Error: cannot update storage credential: Storage credential 'foo' has 0 directly dependent external table(s) and 1 dependent storage location(s); use force option to update anyway.
```

The force option is not exposed via the provider.

### Steps to Reproduce
Change the value of `var.owner` after creating the storage credential and external location in the above example.

### Terraform and provider versions
```
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/databricks/databricks v1.24.1
+ provider registry.terraform.io/gitlabhq/gitlab v16.2.0
+ provider registry.terraform.io/hashicorp/aws v4.51.0
```

The changelog suggests this has not changed on the latest version.

### Debug Output
```
2023-10-06T18:06:16.835-0400 [ERROR] provider.terraform-provider-databricks_v1.24.1: Response contains error diagnostic: tf_rpc=ApplyResourceChange @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR tf_proto_version=5.3 diagnostic_summary="cannot update storage credential: Storage credential 'dev2-audit-logs' has 0 directly dependent external table(s) and 1 dependent storage location(s); use force option to update anyway." tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=2de3931b-3817-d59c-3864-6e48a3dc296d tf_resource_type=databricks_storage_credential timestamp=2023-10-06T18:06:16.834-0400
```

### Important Factoids
This was fixed for `databricks_external_location` in #2546. It was originally brought up in #2188, which also mentioned the issue with `databricks_storage_credential`, but the latter is still an issue - hence this ticket.

I also attempted with `force_destroy = true` and got the same behavior. I confirmed that `force_update = true`, as with the patched external location resource, is *not* exposed on the storage credential resource.

--- a/docs/resources/grant.md
+++ b/docs/resources/grant.md
@@ -211,6 +211,26 @@ resource "databricks_grant" "customers_data_analysts" {
 }
 ```
 
+## Function grants
+
+See [databricks_grants Function grants](grants.md#function-grants) for the list of privileges that apply to Registered models.
+
+```hcl
+resource "databricks_grant" "udf_data_engineers" {
+  function = "main.reporting.udf"
+
+  principal  = "Data Engineers"
+  privileges = ["EXECUTE"]
+}
+
+resource "databricks_grant" "udf_data_analysts" {
+  function = "main.reporting.udf"
+
+  principal  = "Data Analysts"
+  privileges = ["EXECUTE"]
+}
+```
+
 ## Storage credential grants
 
 See [databricks_grants Storage credential grants](grants.md#storage-credential-grants) for the list of privileges that apply to Storage credentials.

--- a/docs/resources/grants.md
+++ b/docs/resources/grants.md
@@ -4,16 +4,15 @@ subcategory: "Unity Catalog"
 # databricks_grants Resource
 
 -> **Note**
-Two different resources help you manage your Unity Catalog grants for a securable. Each of these resources serves a different use case:
-
-[databricks_grants](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/grants): Authoritative. Sets the grants of a securable and replaces any existing grants defined inside or outside of Terraform.
-[databricks_grant](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/grant): Authoritative for a given principal. Updates the grants of a securable to a single principal. Other principals within the grants for the securables are preserved.
-
--> **Note**
   This article refers to the privileges and inheritance model in Privilege Model version 1.0. If you created your metastore during the public preview (before August 25, 2022), you can upgrade to Privilege Model version 1.0 following [Upgrade to privilege inheritance](https://docs.databricks.com/data-governance/unity-catalog/hive-metastore.html)
 
 -> **Note**
   Unity Catalog APIs are accessible via **workspace-level APIs**. This design may change in the future. Account-level principal grants can be assigned with any valid workspace as the Unity Catalog is decoupled from specific workspaces. More information in [the official documentation](https://docs.databricks.com/data-governance/unity-catalog/index.html).
+
+Two different resources help you manage your Unity Catalog grants for a securable. Each of these resources serves a different use case:
+
+- [databricks_grants](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/grants): Authoritative. Sets the grants of a securable and replaces any existing grants defined inside or outside of Terraform.
+- [databricks_grant](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/grant): Authoritative for a given principal. Updates the grants of a securable to a single principal. Other principals within the grants for the securables are preserved.
 
 In Unity Catalog all users initially have no access to data. Only Metastore Admins can create objects and can grant/revoke access on individual objects to users and groups. Every securable object in Unity Catalog has an owner. The owner can be any account-level user or group, called principals in general. The principal that creates an object becomes its owner. Owners receive `ALL_PRIVILEGES` on the securable object (e.g., `SELECT` and `MODIFY` on a table), as well as the permission to grant privileges to other principals.
 
@@ -205,6 +204,25 @@ resource "databricks_grants" "customers" {
   grant {
     principal  = "Data Engineers"
     privileges = ["APPLY_TAG", "EXECUTE"]
+  }
+  grant {
+    principal  = "Data Analysts"
+    privileges = ["EXECUTE"]
+  }
+}
+```
+
+## Function grants
+
+You can grant `ALL_PRIVILEGES` and `EXECUTE` privileges to _`catalog.schema.function`_ specified in the `function` attribute.
+
+```hcl
+resource "databricks_grants" "udf" {
+  function = "main.reporting.udf"
+
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["EXECUTE"]
   }
   grant {
     principal  = "Data Analysts"

--- a/docs/resources/permission_assignment.md
+++ b/docs/resources/permission_assignment.md
@@ -10,26 +10,59 @@ These resources are invoked in the workspace context.
 In workspace context, adding account-level user to a workspace:
 
 ```hcl
-resource "databricks_user" "me" {
+# Use the account provider
+data "databricks_user" "me" {
   user_name = "me@example.com"
+  provider  = databricks.account
 }
 
 resource "databricks_permission_assignment" "add_user" {
-  principal_id = databricks_user.me.id
+  principal_id = data.databricks_user.me.id
   permissions  = ["USER"]
+  provider     = databricks.workspace
 }
 ```
 
 In workspace context, adding account-level service principal to a workspace:
 
 ```hcl
-resource "databricks_service_principal" "sp" {
+# Use the account provider
+data "databricks_service_principal" "sp" {
   display_name = "Automation-only SP"
+  provider     = databricks.account
 }
 
 resource "databricks_permission_assignment" "add_admin_spn" {
-  principal_id = databricks_service_principal.sp.id
+  principal_id = data.databricks_service_principal.sp.id
   permissions  = ["ADMIN"]
+  provider     = databricks.workspace
+}
+```
+
+In workspace context, adding account-level group to a workspace:
+
+```hcl
+# Use the account provider
+data "databricks_group" "account_level" {
+  display_name = "example-group"
+  provider     = databricks.account
+}
+
+# Use the workspace provider
+resource "databricks_permission_assignment" "this" {
+  principal_id = data.databricks_group.account_level.id
+  permissions  = ["USER"]
+  provider     = databricks.workspace
+}
+
+data "databricks_group" "workspace_level" {
+  display_name = "example-group"
+  depends_on   = [databricks_permission_assignment.this]
+  provider     = databricks.workspace
+}
+
+output "databricks_group_id" {
+  value = data.databricks_group.workspace_level.id
 }
 ```
 

--- a/docs/resources/storage_credential.md
+++ b/docs/resources/storage_credential.md
@@ -79,6 +79,7 @@ The following arguments are required:
 - `read_only` - (Optional) Indicates whether the storage credential is only usable for read operations.
 - `skip_validation` - (Optional) Suppress validation errors if any & force save the storage credential.
 - `force_destroy` - (Optional) Delete storage credential regardless of its dependencies.
+- `force_update` - (Optional) Update storage credential regardless of its dependents.
 
 `aws_iam_role` optional configuration block for credential details for AWS:
 

